### PR TITLE
migrate some more "OptionalBool" to "Option[bool]"

### PR DESCRIPTION
**Pull request by 6543**
just some refactoring bits towards replacing **util.OptionalBool** with **optional.Option[bool]**

### CODE REVIEW

File: models/repo/repo_list.go [hunk 1]
```diff
@@ -122,44 +123,44 @@ type SearchRepoOptions struct {
...
-	// True -> include just private
-	// False -> include just public
-	IsPrivate util.OptionalBool
-	// None -> include collaborative AND non-collaborative
-	// True -> include just collaborative
-	// False -> include just non-collaborative
-	Collaborate util.OptionalBool
+	// None[bool]() -> include public and private
+	// Some(true) -> include just private
+	// Some(false) -> include just public
```
Comment (KN4CK3R) [inline models/repo/repo_list.go]: Maybe leave the comment? I think it's still understandable and the new comment is more complex.

File: models/repo/repo_list_test.go [hunk 2]
```diff
@@ -327,30 +327,25 @@ func TestSearchRepository(t *testing.T) {
 						assert.False(t, repo.IsPrivate)
 					}
 
-					if testCase.opts.Fork == util.OptionalBoolTrue && testCase.opts.Mirror == util.OptionalBoolTrue {
+					if testCase.opts.Fork.Value() && testCase.opts.Mirror.Value() {
 						assert.True(t, repo.IsFork || repo.IsMirror)
```
Comment (KN4CK3R) [inline models/repo/repo_list_test.go]: Not related to this PR but shouldn't this be `repo.IsFork && repo.IsMirror`?

File: modules/optional/option_test.go [hunk 3]
```diff
@@ -27,6 +27,16 @@ func TestOption(t *testing.T) {
...
 
+	noneBool := optional.None[bool]()
+	assert.False(t, noneBool.Has())
+	assert.Equal(t, false, noneBool.Value())
+	assert.Equal(t, true, noneBool.ValueOrDefault(true))
+
+	someBool := optional.Some(true)
+	assert.True(t, someBool.Has())
+	assert.Equal(t, true, someBool.Value())
+	assert.Equal(t, true, someBool.ValueOrDefault(false))
```
Comment (KN4CK3R) [inline modules/optional/option_test.go]:
```suggestion
	assert.False(t, noneBool.Value())
	assert.True(t, noneBool.ValueOrDefault(true))

	someBool := optional.Some(true)
	assert.True(t, someBool.Has())
	assert.True(t, someBool.Value())
	assert.True(t, someBool.ValueOrDefault(false))
```

File: models/repo/repo_list.go (hunk 1 above)
Comment (6543) [reply models/repo/repo_list.go]: reverted

File: models/repo/repo_list_test.go (hunk 2 above)
Comment (6543) [inline models/repo/repo_list_test.go]:
```suggestion
						assert.True(t, repo.IsFork && repo.IsMirror)
```

Comment (6543) [reply models/repo/repo_list_test.go]: well you might have spotted right ... let's just fix it and see if tests fail :)

if not then let it fixed and move on

**Review by KN4CK3R — APPROVED** [review]

**Review by lunny — APPROVED** [review]

**Pull request merged**